src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -121,23 +127,38 @@ export class AlertService {
      *                   If `translateKey` is available then it's translation else `message` is used for showing.
      * @returns  Added alert
      */
-    addAlert(alert: AlertBase): Alert {
-        const alertInternal = alert as AlertInternal;
-        alertInternal.id = this.alertId++;
+    addAlert(alert: AlertCreationProperties): Alert {
+        // Defensive copy to prevent overwrites
+        const alertInternal = { ...alert } as AlertInternal;
         alertInternal.isOpen = true;
 
-        if (alertInternal.translationKey) {
-            // in case a translation key is defined, we use it to create the message
-            const translatedMessage = this.translateService.instant(alertInternal.translationKey, alertInternal.translationParams);
-            // if translation key exists
-            if (translatedMessage !== `${translationNotFoundMessage}[${alertInternal.translationKey}]`) {
-                alertInternal.message = translatedMessage;
-            } else if (!alertInternal.message) {
-                alertInternal.message = alertInternal.translationKey;
+        if (!alert.disableTranslation) {
+            if (alert.translationKey) {
+                // in case a translation key is defined, we use it to create the message
+                const translatedMessage = this.translateService.instant(alert.translationKey, alert.translationParams);
+                // if translation key exists
+                if (translatedMessage !== `${translationNotFoundMessage}[${alert.translationKey}]`) {
+                    alertInternal.message = translatedMessage;
+                } else if (!alertInternal.message) {
+                    alertInternal.message = alert.translationKey;
+                }
+            } else if (alertInternal.message) {
+                // Note: in most cases, our code passes the translation key as message
+                alertInternal.message = this.translateService.instant(alertInternal.message, alert.translationParams);
+            }
+
+            if (alertInternal.message?.startsWith('translation-not-found')) {
```

Comment:
Isn't this one here just the else clause of line 140 conditional?

Reply:
I'll rewrite the logic a bit to merge the two blocks.